lib/mix/tasks/nerves.gen.script.upload.ex

```diff
@@ -0,0 +1,24 @@
+defmodule Mix.Tasks.Nerves.Gen.Script.Upload do
+  @shortdoc "Generates a script for custom remote firmware updates"
+
+  use Mix.Task
+  def run(_) do
+    Mix.shell.info """
+
+    Generate a new custom script to use instead of firmware.push
+    """
+    filename = Mix.shell.prompt("Enter file to use for your new script (./upload.sh):")
+      |> String.strip
+    if filename == "" do
+      filename = "upload.sh"
+    end
+    upload_script_contents = EEx.eval_file("./priv/templates/script.upload.eex", [])
+    Mix.shell.info """
+
+    Generating a new file named #{filename}
+    """
+    File.write(filename, "#{upload_script_contents}\n")
+
+    System.cmd("chmod", ["+x", filename])
```

Comment:
Could you change this to use `File.chmod!(filename, 0o755)`?

Reply:
Yep, I can do that. I'm also seeing a warning when I run so I'll try and clean that up as well.